src/zenml/artifacts/utils.py

```diff
@@ -577,6 +636,146 @@ def get_artifacts_versions_of_pipeline_run(
 # -------------------------
 
 
+def _check_if_artifact_with_given_uri_already_registered(
+    artifact_store: "BaseArtifactStore",
+    uri: str,
+    name: str,
+) -> None:
+    """Check if the given artifact store already contains an artifact with the given URI.
+
+    Args:
+        artifact_store: The artifact store to check.
+        uri: The uri of the artifact.
+        name: The name of the artifact.
+
+    Raises:
+        RuntimeError: If the artifact store already contains an artifact with
+            the given URI.
+    """
+    if artifact_store.exists(uri):
+        # This check is only necessary for manual saves as we already check
+        # it when creating the directory for step output artifacts
+        other_artifacts = Client().list_artifact_versions(uri=uri, size=1)
+        if other_artifacts and (other_artifact := other_artifacts[0]):
+            raise RuntimeError(
+                f"Cannot create new artifact {name} version with URI "
```

Comment:
I forgot to mention this here. Since you moved the function out, we might want to rephrase the error message here. Previously this was only used in a context where we were trying to create an artifact, now it is a check that can be used in different settings as well.

Reply:
I would rather disagree here: the context is still the same - we try to create AV in ZenML on a given URI and it fails to do so since URI is already taken. I'd prefer to keep this one as is.